code/game/objects/heatwave.dm

```diff
@@ -0,0 +1,49 @@
+proc/heatwave(turf/epicenter, heavy_range, light_range, damage, fire_stacks, penetration, log=0)
+	if(!epicenter) return
+
+	if(!istype(epicenter, /turf))
```

Comment:
I think you can just put `!istype(epicenter)` to do the same thing since it checks if epicenter has the type it was declared with (`turf/epicenter` here). But you can let it like that.